examples/drosophila/preprocessing_wing-disc.py

```diff
@@ -0,0 +1,239 @@
+            str(sample_folder.name)
+            in right_wingdisc_catalog["filename"].astype(str).tolist()
+        ):
+            image_array = np.flip(image_array, axis=2)
+            logger.info(f"Mirrored {sample_folder.name}.")
+        else:
+            logger.info(f"{sample_folder.name} is not right wing disc.")
+            image_array = image_array
+
+        # Downsample the image array
+        logger.info(f"Downsampling {sample_folder}...")
+
+        image_dask = da.from_array(image_array, chunks={0: 1, 1: -1, 2: -1})
+        down_sampled_image = resize_anisotropic_image_stack(
+            image_dask, in_plane_factor, axial_factor
+        )
+        down_sampled_image = img_as_uint(down_sampled_image, force_copy=False)
+
+        # Apply median filter to the downsampled image
+        mf_image = median_filter(down_sampled_image, size=3)
+        mf_image = np.pad(mf_image, ((15, 15), (0, 0), (0, 0)), mode="reflect")
+        mf_image = mf_image * (mf_image > 104)
+        logger.debug(
+            "Applied median filter and padding to the downsampled image and set the threshold as 104"
+        )
+
+        # Save the downsampled and filtered image as tif
+        saving_folder = (
+            template_raw_data
+            / f"{source_data.name}"
+            / downsampled_filename.split(".")[0]
+        )
+        saving_folder.mkdir(exist_ok=True, parents=True)
+        assert Path(
+            saving_folder
+        ).exists(), f"Filepath {saving_folder} not found"
+        saving_path = saving_folder / downsampled_filename
+        save_any(mf_image, saving_path)
+        logger.info(
+            f"{sample_folder} downsampled, filtered and padded, saved as {downsampled_filename}"
+        )
+
+        # Save the downsampled,filtered and padded image as nifti
+        nii_path = file_path_with_suffix(
+            saving_path,
+            "_downsampled_filtered_padded_normalized",
+            new_ext=".nii.gz",
+        )
+        vox_sizes = [
+            target_isotropic_resolution,
+        ] * 3
+        save_as_asr_nii(mf_image, vox_sizes, nii_path)
+        logger.info(
+            f"Saved downsampled and filtered image as {nii_path.name}."
+        )
+
+        # Generate the wingdisc mask
+        image_ants = ants.image_read(nii_path.as_posix())
+        mask_data = create_mask(
+            image_ants.numpy(),
+            gauss_sigma=1,
+            threshold_method="triangle",
+            closing_size=3,
+        )
+        mask_path = file_path_with_suffix(nii_path, "_mask")
+        mask = image_ants.new_image_like(mask_data.astype(np.uint8))
+
+        ants.image_write(mask, mask_path.as_posix())
+        logger.debug(
+            f"Generated brain mask with shape: {mask.shape} "
+            f"and saved as {mask_path.name}."
+        )
+
+        # Normalized the image
+        normalised_image = (mf_image - np.min(mf_image)) / (
+            np.max(mf_image) - np.min(mf_image)
+        )
```

Comment:
Think we are "squishing" the range of pixels too much here for our new images?
```suggestion
        normalised_image = (mf_image - np.min(mf_image)) / ((
            np.max(mf_image) - np.min(mf_image)*65535
        )
```
maybe? (to limit to unsigned int 16)

Or we can try using [the bias correction algorithm we used for the bird template](https://github.com/brainglobe/brainglobe-template-builder/blob/73238dfc109a145aca66717d8d0e3764c826a07b/examples/blackcap/2_prep_lowres.py#L141) instead of this.